src/alloc/abort.rs

```diff
@@ -0,0 +1,126 @@
+#![allow(clippy::use_self)]
+
+use liballoc::alloc::handle_alloc_error;
+
+use crate::alloc::{
+    AllocRef,
+    BuildAllocRef,
+    DeallocRef,
+    Global,
+    NonZeroLayout,
+    NonZeroUsize,
+    ReallocRef,
+    System,
+};
+use core::ptr::NonNull;
+
+/// Indicates that the allocator is allowed to be used where we panic on OOM.
+pub trait Abort: AllocRef {}
+
+impl Abort for Global {}
+impl Abort for System {}
```

Comment:
`System` needs std support:

```suggestion
#[cfg(feature = "std")]
impl Abort for std::alloc::System {}
```